src/librustc_mir/borrow_check/nll/region_infer/error_reporting/mod.rs

```diff
@@ -120,19 +122,50 @@ impl<'tcx> RegionInferenceContext<'tcx> {
         // we still want to screen for an "interesting" point to
         // highlight (e.g., a call site or something).
         let target_scc = self.constraint_sccs.scc(target_region);
-        let best_choice = (0..path.len()).rev().find(|&i| {
-            let constraint = path[i];
+        let mut range = 0..path.len();
+
+        let should_reverse = match from_region_origin {
+            NLLRegionVariableOrigin::FreeRegion
+                | NLLRegionVariableOrigin::Existential { was_placeholder: false  } => {
+                    true
+            }
+            NLLRegionVariableOrigin::Placeholder(_)
+                | NLLRegionVariableOrigin::Existential { was_placeholder: true  } => {
```

Comment:
I'm going to test this myself, but maybe you know the answer -- what changes when `was_placeholder` is ignored here? which tests?

Reply:
Some of the new tests should start failing. I can't remember which of them actually triggered this logic, but any test that causes us to replace a `NLLRegionVariableOrigin::Placeholder` with a `NLLRegionVariableOrigin::Existential` will have an incorrect error span.